pkg/volume/plugins.go

```diff
@@ -119,6 +119,39 @@ type VolumePluginMgr struct {
 	plugins map[string]VolumePlugin
 }
 
+// InternalVolume is an internal representation of a volume.  All API volume types translate to InternalVolume.
+type InternalVolume struct {
+	Name string
+	InternalVolumeSource
+}
+
+// InternalVolumeSource is a union structure encompassing all types of volumes.
+// This union allows us to separate API types but treat all volumes equally as plugins.
+type InternalVolumeSource struct {
+	api.VolumeSource
+	api.PersistentVolumeSource
+}
+
+func ConvertFromVolume(vs *api.Volume) *InternalVolume {
+	return &InternalVolume{
+		Name: vs.Name,
+		InternalVolumeSource: InternalVolumeSource{
+			vs.VolumeSource,
+			api.PersistentVolumeSource{},
```

Comment:
don't init the field you don't need - it's just noise.  Use a named init here and it will be clearer